### DISCUSSION TITLE
Ability to override behavior at exactly one dimension above a record

### DISCUSSION THREAD

**Discussion by jpivarski** [scikit-hep/awkward#1660] [Deprecations]
There are two ways to overload an array that contains a named record:

```python
ak._v2.behavior["*", "NameOfRecord"] = ClassForArray
ak._v2.behavior[".", "NameOfRecord"] = ClassForArray
```

The first of these overrides any kind of list that is surrounding the records, so that you can define vectorized operations.

The second only overrides one dimension above the records.

The second case was broken, never tested, and no one has ever asked me about it. In [#1651](https://github.com/scikit-hep/awkward/pull/1651), I dropped it for v2 with the idea that we can reintroduce it correctly if asked.

What do you think? Does anybody need the `"."` method?

@nsmith-, is this used in NanoEvents?

**Comment by nsmith-** [comment]
NanoEvents always uses `awkward.mixin_class` to define classes, and that uses only the `"*"` method: https://github.com/scikit-hep/awkward/blob/15318e1d40f3578c55195e6928e8c2b37d948071/src/awkward/behaviors/mixins.py#L46
